### PR TITLE
Fix: Add --dry-run to the tool for applying state-parts

### PR DESCRIPTION
**Pull request by nikurt**
Also rework the state parts command API to be a single command with two subcommands instead of two separate commands.

### CODE REVIEW

**Comment by akhi3030** [comment]
I think @marcelo-gonzalez might be a good reviewer for this PR.  As they are not a codeowner, I am happy to approve once they do the review.

**Review by marcelo-gonzalez — APPROVED** [review]

File: tools/state-viewer/src/state_parts.rs [hunk 1]
```diff
@@ -19,6 +19,79 @@ use std::str::FromStr;
...
+                    store,
+                    Location::new(root_dir, (s3_bucket, s3_region)),
+                );
+            }
+            StatePartsSubCommand::Dump { part_from, part_to, epoch_selection } => {
+                dump_state_parts(
+                    epoch_selection,
+                    shard_id,
+                    part_from.or(part_id),
+                    part_to.or(part_id.map(|x| x + 1)),
```
[the remark this answers is not in the document]
Comment (nikurt) [reply tools/state-viewer/src/state_parts.rs:84]: Agreed. Done.

File: tools/state-viewer/src/cli.rs [hunk 2]
```diff
@@ -27,8 +26,6 @@ pub enum StateViewerSubCommand {
     /// even if it's not included in any block on disk
     #[clap(alias = "apply_receipt")]
     ApplyReceipt(ApplyReceiptCmd),
-    /// Apply all or a single state part of a shard.
-    ApplyStateParts(ApplyStatePartsCmd),
```
[the remark this answers is not in the document]
Comment (nikurt) [reply tools/state-viewer/src/cli.rs:31]: Mentioned in the `CHANGELOG`.
I can only imagine StateParts and StateChanges commands to be useful to somebody working on State Sync. Very unlikely to break anyone.

File: tools/state-viewer/src/state_parts.rs [hunk 3]
```diff
@@ -19,6 +19,79 @@ use std::str::FromStr;
...
+        #[clap(long)]
+        dry_run: bool,
+        /// If provided, this value will be used instead of looking it up in the headers.
+        /// Use if those headers or blocks are not available.
+        #[clap(long)]
+        state_root: Option<StateRoot>,
+
+        /// Selects an epoch. The dump will be of the state at the beginning of this epoch.
+        #[clap(subcommand)]
+        epoch_selection: EpochSelection,
```
[the remark this answers is not in the document]
Comment (nikurt) [reply tools/state-viewer/src/state_parts.rs:36]: Would be nice, but `clap` doesn't allow more than 1 subcommand per command.
Have to go with a common sub-sub-command.

**Review by akhi3030 — APPROVED** [review]

File: tools/state-viewer/src/state_parts.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (marcelo-gonzalez) [reply tools/state-viewer/src/state_parts.rs:38]: ah no i meant like as one of the options in `struct StatePartsCmd`. not a huge deal though obviously

**Pull request merged**